### PR TITLE
git versions: fix commit shas [automated]

### PR DESCRIPTION
**Pull request by haampie**
A few commit shas are missing or wrong.

### CODE REVIEW

File: var/spack/repos/builtin/packages/hpx/package.py [hunk 1]
```diff
@@ -21,7 +21,7 @@ class Hpx(CMakePackage, CudaPackage, ROCmPackage):
     tags = ["e4s"]
 
     version("master", branch="master")
-    version("stable", tag="stable", commit="38d5bf935e5a49f9466c5e615e04e8d553a73dc6")
+    version("stable", tag="stable", commit="103a7b8e3719a0db948d1abde29de0ff91e070be")
```
Comment (haampie) [inline var/spack/repos/builtin/packages/hpx/package.py:24]: Is this useful to keep? Might as well remove it if the stable tag isn't ... stable :smirk:

File: var/spack/repos/builtin/packages/ipm/package.py [hunk 2]
```diff
@@ -19,7 +19,7 @@ class Ipm(AutotoolsPackage):
     maintainers("Christoph-TU")
 
     version("master", branch="master", preferred=True)
-    version("2.0.6", tag="2.0.6")
+    version("2.0.6", tag="2.0.6", commit="b008141ee16d39b33e20bffde615564afa107575")
```
Comment (Christoph-TU) [inline var/spack/repos/builtin/packages/ipm/package.py:22]: Confirmed that the commit sha is correct and matches the tag

**Review by edwardhartnett — APPROVED** [review]

**Review by hariharan-devarajan — APPROVED** [review]
The commit hash for cpp-logger is correct.

File: var/spack/repos/builtin/packages/exago/package.py [hunk 3]
```diff
@@ -17,7 +17,9 @@ class Exago(CMakePackage, CudaPackage, ROCmPackage):
     git = "https://github.com/pnnl/ExaGO.git"
     maintainers("ryandanehy", "cameronrutherford", "pelesh")
 
-    version("1.5.1", tag="v1.5.1", submodules=True)
+    version(
+        "1.5.1", tag="v1.5.1", commit="84e9faf9d9dad8d851075eba26038338d90e6d3a", submodules=True
+    )
     version("1.5.0", commit="227f49573a28bdd234be5500b3733be78a958f15", submodules=True)
     version("1.4.1", commit="ea607c685444b5f345bfdc9a59c345f0f30adde2", submodules=True)
     version("1.4.0", commit="4f4c3fdb40b52ace2d6ba000e7f24b340ec8e886", submodules=True)
```
Comment (cameronrutherford) [inline var/spack/repos/builtin/packages/exago/package.py:25]:
```suggestion
    version(
        "1.5.1", tag="v1.5.1", commit="84e9faf9d9dad8d851075eba26038338d90e6d3a", submodules=True
    )
    version(
        "1.5.0", tag="v1.5.0", commit="227f49573a28bdd234be5500b3733be78a958f15", submodules=True
    )
    version(
        "1.4.1", tag="v1.4.1", commit="ea607c685444b5f345bfdc9a59c345f0f30adde2", submodules=True
    )
    version(
        "1.4.0", tag="v1.4.0", commit="4f4c3fdb40b52ace2d6ba000e7f24b340ec8e886", submodules=True
    )
```

@eugeneswalker noticed the commit hash was wrong for 1.5.1 in https://github.com/spack/spack/pull/40676, so not sure how/when that went awry...

All versions have a matching tag, and they are intended to be stable tags, so I think it makes sense too keep track of both for all exago versions. Would be good to fix this/apply to remaining versions in ExaGO 1.6.0 release https://github.com/spack/spack/pull/40614 @ryandanehy

Comment (haampie) [reply var/spack/repos/builtin/packages/exago/package.py:25]: yeah, keeping track of the commit sha too is exactly there to guard against accidental/malicious retagging of commits.

**Review by cameronrutherford — APPROVED** [review]

**Review by AlexanderRichert-NOAA — APPROVED** [review]
approve for upp; all three commit hashes are good

File: var/spack/repos/builtin/packages/neovim/package.py [hunk 4]
```diff
@@ -16,7 +16,7 @@ class Neovim(CMakePackage):
     maintainers("albestro", "trws")
 
     version("master", branch="master")
-    version("stable", tag="stable", commit="7d4bba7aa7a4a3444919ea7a3804094c290395ef")
+    version("stable", tag="stable", commit="d772f697a281ce9c58bf933997b87c7f27428a60")
```
Comment (albestro) [inline var/spack/repos/builtin/packages/neovim/package.py:19]: ✅ verified, looks correct now. thanks

**Review by jcbowden — COMMENTED** [review]
Damaris additions look fine.

**Pull request merged**